web/src/main/java/org/springframework/security/web/server/authentication/logout/DelegatingServerLogoutHandler.java

```diff
@@ -35,21 +37,34 @@
  * @since 5.1
  */
 public class DelegatingServerLogoutHandler implements ServerLogoutHandler {
-	private final List<ServerLogoutHandler> delegates;
+	private final List<ServerLogoutHandler> delegates = new ArrayList<>();
 
 	public DelegatingServerLogoutHandler(ServerLogoutHandler... delegates) {
 		Assert.notEmpty(delegates, "delegates cannot be null or empty");
-		this.delegates = Arrays.asList(delegates);
+		this.delegates.addAll(Arrays.asList(delegates));
 	}
 
-	public DelegatingServerLogoutHandler(List<ServerLogoutHandler> delegates) {
+	public DelegatingServerLogoutHandler(Collection<ServerLogoutHandler> delegates) {
 		Assert.notEmpty(delegates, "delegates cannot be null or empty");
-		this.delegates = new ArrayList<>(delegates);
+		this.delegates.addAll(delegates);
+	}
+
+	/**
+	 * Adds a delegate {@link ServerLogoutHandler}
+	 *
+	 * @param delegate The delegate to add
+	 */
+	public void addDelegate(ServerLogoutHandler delegate) {
```

Comment:
We want this to be immutable, so this method needs to be removed

Reply:
If we truly want this to be immutable, then the configuration DSL could potentially be setting a `DelegatingServerLogoutHandler` who has in its collection of delegates other `DelegatingServerLogoutHandler`s (which could, in turn, contain other `DelegatingServerLogoutHandler`s). This method's purpose was to allow the configuration DSL to guard against that.

If you're ok with that situation then sure we can remove this method.

Reply:
I think the DSL would just track the `ServerLogoutHandler`s If there were multiple it would create a `DelegatingServerLogoutHandler` with all the added `ServerLogoutHandler`s. Otherwise it would just use the single `ServerLogoutHandler`